src/jit/hwintrinsicxarch.cpp

```diff
@@ -1044,6 +1045,114 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
+            assert((baseType >= TYP_BYTE) && (baseType <= TYP_DOUBLE));
+            assert(baseType != TYP_FLOAT);
+
+            op1          = impPopStack().val;
+            GenTree* src = nullptr;
+
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                {
+                    src     = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_SSE2_SetScalarVector128, baseType, simdSize);
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, gtCloneExpr(src), gtNewIconNode(0),
+                                                       NI_SSE2_Shuffle, baseType, simdSize);
+                    break;
+                }
+
+                case TYP_LONG:
+                case TYP_ULONG:
+                {
+                    var_types shuffleBaseType = TYP_UNKNOWN;
+                    if (baseType == TYP_LONG)
+                    {
+                        intrinsic       = NI_SSE2_ConvertScalarToVector128Int64;
+                        shuffleBaseType = TYP_INT;
+                    }
+                    else
+                    {
+                        intrinsic       = NI_SSE2_ConvertScalarToVector128UInt64;
+                        shuffleBaseType = TYP_UINT;
+                    }
+
+                    src     = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, intrinsic, baseType, simdSize);
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, gtNewIconNode(0b01000100), NI_SSE2_Shuffle,
+                                                       shuffleBaseType, simdSize);
+                    break;
+                }
+
+                case TYP_INT:
+                case TYP_UINT:
+                {
+                    src = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_SSE2_ConvertScalarToVector128Int32, TYP_INT,
+                                                   simdSize);
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, gtNewIconNode(0), NI_SSE2_Shuffle, baseType,
+                                                       simdSize);
+                    break;
+                }
+
+                case TYP_SHORT:
+                case TYP_USHORT:
+                {
+                    GenTree* srcClone = nullptr;
+
+                    // Initialize XMM register with initial value
+                    src = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_SSE2_ConvertScalarToVector128Int32, TYP_INT,
+                                                   simdSize);
+
+                    // src is used twice - clone it
+                    CORINFO_CLASS_HANDLE vectorHandle = (baseType == TYP_SHORT) ? Vector128ShortHandle : Vector128UShortHandle;
+                    src = impCloneExpr(src, &srcClone, vectorHandle, (unsigned)CHECK_SPILL_ALL,
+                        nullptr DEBUGARG("SetAllVector - duplicate src tree with initialized XMM register"));
+
+                    GenTree* tmp = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, srcClone, NI_SSE2_UnpackLow,
```

Comment:
We can change `NI_SSE2_UnpackLow`'s `numArg` field to `-1` to let `UnpackLow` accept 1-arg (internally), which avoids the `impCloneExpr` here.

BTW, the backend supports 1-arg from codgen for `punpack*`.